frontend/src/page/Realm.tsx

```diff
@@ -1,20 +1,49 @@
 import { jsx } from "@emotion/core";
 import React from "react";
-import { match } from "react-router-dom";
+import { Link, match } from "react-router-dom";
+
+import { Breadcrumbs } from "../ui/Breadcrumbs";
 
 
 type Props = {
     match: match<{ path?: string }>,
 };
 
 export const Realm: React.FC<Props> = ({ match }) => {
-    // const { path } = useParams();
-    const path = (match.params.path || "").split("/");
+    const isRoot = !match.params.path;
+    const currentPath = match.params.path || "";
+    const ids = resolvePath(currentPath);
+    if (ids == null) {
+        return <b>Realm path not found :(</b>;
+    }
+
+    const realmId = ids[ids.length - 1];
+    const realm = REALMS[realmId];
+
+    const breadcrumbs = [];
+    let path = "/r";
+    for (const id of ids.splice(1)) {
+        path += "/" + REALMS[id].path;
+        breadcrumbs.push({
+            label: REALMS[id].name,
+            href: path,
+        });
+    }
+
+    const pathBase = "/r/" + (currentPath && currentPath + "/");
```

Comment:
```suggestion
    const pathBase = `/r/${currentPath}/`;
```

`currentPath` should not be optional, and even now that it is, you set it to `""` when it's falsy above.

Reply:
But your code would result in `"/r//"` if `currentPath` is empty.

Reply:
Ha, true! Nevermind, then. :sweat_smile: